types/vote_set.go

```diff
@@ -128,20 +128,29 @@ func (voteSet *VoteSet) addVote(val *Validator, valIndex int, vote *Vote) (bool,
 		return false, 0, ErrVoteUnexpectedStep
 	}
 
+	var doubleSign bool
 	// If vote already exists, return false.
 	if existingVote := voteSet.votes[valIndex]; existingVote != nil {
 		if bytes.Equal(existingVote.BlockHash, vote.BlockHash) {
 			return false, valIndex, nil
 		} else {
-			// Check signature.
-			if !val.PubKey.VerifyBytes(SignBytes(config.GetString("chain_id"), vote), vote.Signature) {
```

Comment:
We should still check the signature, I think, before returning ErrVoteConflictingSignature (e.g. otherwise we're blaming the validator when we should be blaming the peer)